packages/core/entity/FileDetail/RenderZarrThumbnailURL.ts

```diff
@@ -38,89 +38,129 @@ function transformAxes(originalArray: AxisData[]): TransformedAxes[] {
         value: item.type !== "space" ? 0 : null,
     }));
 }
+/**
+ * Helper function to handle retry logic with timeout for async operations.
+ * It retries the operation up to the specified number of times and aborts if it takes too long.
+ */
+async function retryWithTimeout<T>(fn: () => Promise<T>, retries = 3, timeout = 5000): Promise<T> {
+    let attempt = 0;
+    while (attempt < retries) {
+        try {
+            return await withTimeout(fn(), timeout);
```

Comment:
What is the purpose of the timeout here. Does it help the callstack avoid getting overloaded somehow?

Reply:
From what i can tell some async requests in the zarr render dont register as  failures, This just says if its taking too long count it as a failure.